katportalclient/test/test_client.py

```diff
@@ -1565,44 +1722,86 @@ def mock_async_fetchers(valid_responses, invalid_responses, starts_withs=None,
     assert(len(starts_withs) == num_calls)
     assert(len(ends_withs) == num_calls)
     assert(len(containses) == num_calls)
-    mock_fetches = [mock_async_fetcher(v, i, s, e, c)
+    mock_fetches = [self.mock_async_fetcher(v, i, s, e, c)
                     for v, i, s, e, c in zip(
                         valid_responses, invalid_responses,
                         starts_withs, ends_withs, containses)]
     # flip order so that poping effectively goes from first to last input
     mock_fetches.reverse()
 
-    def mock_fetch(url):
-        single_fetch = mock_fetches.pop()
-        return single_fetch(url)
+        def mock_fetch(url):
+            if url == SITEMAP_URL:
+                # Don't consume from the list, because it's not the request
+                # we're looking for.
+                single_fetch = mock_fetches[-1]
+            else:
+                single_fetch = mock_fetches.pop()
+            return single_fetch(url)
 
-    return mock_fetch
+        return mock_fetch
 
 
 def mock_async_fetcher(valid_response, invalid_response, starts_with=None,
```

Comment:
`mock_async_fetcher` must be changed to a class method - see code in master branch.